srcCpp/RTIDDSImpl.cxx

```diff
@@ -2040,58 +1635,93 @@ bool RTIDDSImpl<T>::Initialize(int argc, char *argv[])
             qos,
             _ProfileLibraryName,
             "BaseProfileQos") != DDS_RETCODE_OK) {
-        fprintf(
-                stderr,
+        fprintf(stderr,
                 "Problem setting QoS Library \"%s::BaseProfileQos\" for participant_qos.\n",
                 _ProfileLibraryName);
     }
 
+    // Set initial peers and not use multicast
+    if (_peer_host_count > 0) {
+        printf("Initial peers:\n");
+        for (int i = 0; i < _peer_host_count; ++i) {
+            printf("\t%s\n", _peer_host[i]);
+        }
+        qos.discovery.initial_peers.from_array(
+                (const char **)_peer_host,
+                _peer_host_count);
+        qos.discovery.multicast_receive_addresses.length(0);
+    }
+
+    // Configure security
   #ifdef RTI_SECURE_PERFTEST
-    if (_secureUseSecure) {
+    if (_security.useSecurity) {
         // validate arguments
-        if (!validateSecureArgs()) {
+        if (!_security.validateSecureArgs(_isPublisher)) {
             fprintf(stderr, "Failed to configure security plugins\n");
             return false;
         }
         // configure
-        if (!configureSecurePlugin(qos)) {
+        if (!PerftestConfigureSecurity(_security, qos)) {
             fprintf(stderr, "Failed to configure security plugins\n");
             return false;
         }
     }
-  #endif
+  #endif // RTI_SECURE_PERFTEST
 
-    // set initial peers and not use multicast
-    if ( _peer_host_count > 0 ) {
-        printf("Initial peers:\n");
-        for ( int i = 0; i< _peer_host_count; ++i) {
-            printf("\t%s\n", _peer_host[i]);
-        }
-        qos.discovery.initial_peers.from_array(
-            (const char **)_peer_host,
-            _peer_host_count);
-        qos.discovery.multicast_receive_addresses.length(0);
+    if (_AutoThrottle) {
+        DDSPropertyQosPolicyHelper::add_property(
```

Comment:
Why do not we check retcode here. 

```
        retcode = DDSPropertyQosPolicyHelper::add_property(
                qos.property,
                "dds.domain_participant.auto_throttle.enable",
                "true",
                false);
        if (retcode != DDS_RETCODE_OK) {
            printf("Failed to add property "
                    "dds.domain_participant.auto_throttle.enable\n");
            return false;
        }
```
And can you do it in the all the add_property calls?

Reply:
Done.